core-tests/src/test/java/org/mule/runtime/core/el/context/MessageContextTestCase.java

```diff
@@ -75,15 +78,15 @@ public void correlationId() throws Exception {
 
   @Test
   public void correlationSequence() throws Exception {
-    when(event.getGroupCorrelation().getSequence()).thenReturn(of(4));
+    when(event.getGroupCorrelation()).thenReturn(Optional.of(GroupCorrelation.of(4)));
     assertEquals(4, evaluate("message.correlationSequence", event));
     assertFinalProperty("message.correlationSequence=2", event);
   }
 
   @Test
   public void correlationGroupSize() throws Exception {
-    when(event.getGroupCorrelation()).thenReturn(new GroupCorrelation(null, 4));
-    assertEquals(4, evaluate("message.correlationGroupSize", event));
+    when(event.getGroupCorrelation()).thenReturn(Optional.of(GroupCorrelation.of(0, 2)));
+    assertEquals(2, evaluate("message.correlationGroupSize", event));
```

Comment:
Because previously there was a bug in implementation, and sequenceId/groupSize were mixed up. Best way to avoid this is to use different test values in test case.